### PR TITLE
Added netbox plugin

### PR DESCRIPTION
**Pull request by insomniacslk**
NetBox is an open source network management system. This plugin allows
using NetBox to assign IP addresses. https://netbox.readthedocs.io/ .

Signed-off-by: Andrea Barberio <insomniac@slackware.it>

### CODE REVIEW

**Comment by insomniacslk** [comment]
Currently v4-only, so do not merge yet

**Comment by insomniacslk** [comment]
# testing

## server side
On NetBox: configured a device with an interface whose primary IP is 10.10.10.10/24:
```
$ curl -H 'Authorization: Token my_api_token' -H "Accept: application/json; indent=4" http://10.10.10.1/api/dcim/devices/1/  2> /dev/null | jq '.primary_ip'
{
  "id": 2,
  "url": "http://10.10.10.1/api/ipam/ip-addresses/2/",
  "family": 4,
  "address": "10.10.10.10/24"
}

```

config.yml
```
server4:
    interface: "enp2s0"
    plugins:
        - lease_time: 60s
        - server_id: 10.0.0.1
        - dns: 8.8.8.8 8.8.4.4
        - router: 10.0.0.1
        - netbox: http://10.10.10.1 my_api_key
        - netmask: 255.255.255.0
```
server output
```
2020/01/02 10:30:26 Handling request from 0.0.0.0:68                            
[2020-01-02T10:30:26Z]  INFO plugins/netbox: received DHCPv4 packet: DHCPv4 Message                                                                             
  opcode: BootRequest                                                           
  hwtype: Ethernet                                                              
  hopcount: 0                                                                   
  transaction ID: 0x022a7e05
  num seconds: 0                                                       [0/19774]
  flags: Unicast (0x00)
  client IP: 0.0.0.0   
  your IP: 0.0.0.0  
  server IP: 0.0.0.0
  gateway IP: 0.0.0.0
  client MAC: aa:bb:cc:dd:ee:ff
  server hostname:
  bootfile name:
  options:
    Host Name: test-device
    Requested IP Address: 10.10.10.10
    DHCP Message Type: REQUEST
    Parameter Request List: Subnet Mask, Time Offset, Router, Domain Name Server
, Host Name, Domain Name, Interface MTU, Broadcast Address, NTP Servers, NetBIOS
 over TCP/IP Name Server, NetBIOS over TCP/IP Scope, DNS Domain Search List, Cla
ssless Static Route

[2020-01-02T10:30:26Z]  INFO plugins/netbox: Checking MAC 00:B5:6D:03:00:06
```

## client side
```
$ sudo dhclient -v -4 eth0
[...]
Listening on LPF/eth0/aa:bb:cc:dd:ee:ff
Sending on   LPF/eth0/v
Sending on   Socket/fallback
DHCPREQUEST of 10.10.10.10 on eth0 to 255.255.255.255 port 67 (xid=0x022a7e05)
DHCPACK of 10.10.10.127 from 10.10.10.1
/tmp/tmp.Dn0KDgVoxR /tmp/tmp.B0TlyXmuNo differ: char 1, line 1
bound to 10.10.10.127 -- renewal in 26 seconds.
```

File: netbox/netbox.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type addr struct {
```
Comment (pmazzini) [inline netbox/netbox.go]: Can this be replaced with net.IPNet?

https://golang.org/pkg/net/#IPNet

Comment (insomniacslk) [reply netbox/netbox.go]: I couldn't remember the name and spent 20 minutes trying to find it.... thanks

Comment (insomniacslk) [reply netbox/netbox.go]: done

**Review by pmazzini — APPROVED** [review]

**Comment by insomniacslk** [comment]
added v6 support

**Pull request merged**